### PR TITLE
Add documentation for optional and not_exported

### LINKED ISSUE #1020: Document optional fields in the manual
Optional fields have landed in `0.3` but are nowhere documented. They should be mentioned in the user manual, in both the syntax section and the merging section.

### PR DESCRIPTION
**Pull request by yannham**
Closes #1020. Document `not_exported` and `optional` fields in the manual.

### CODE REVIEW

File: doc/manual/syntax.md [hunk 1]
```diff
@@ -567,3 +569,35 @@ Examples:
...
+```text
+> let value = { foo = 1, bar | not_exported = 2}
+> value
+{ foo = 1, bar = 2 }
+
+> builtin.serialize `Json value
+"{
+  "foo": 1
+}"
+```
```
Comment (vkleen) [inline doc/manual/syntax.md:603]: The metadata section of the manual might need a more comprehensive overhaul, since we now have a distinction between `|` at record fields and `|` as contract assertion for values. But that's probably better done in another PR down the line.

**Review by vkleen — APPROVED** [review]

File: doc/manual/syntax.md (hunk 1 above)
Comment (yannham) [reply doc/manual/syntax.md:603]: Good point. Let's open an issue for that.

File: doc/manual/merging.md [hunk 2]
```diff
@@ -257,6 +259,90 @@ Metadata can be attached to values thanks to the `|` operator. Metadata
...
 
+### Optional fields
+
+A field can be marked as optional using the `optional` annotation:
+
+```nickel
+# schema.ncl
+{
+  Command = {
+    commad
```
Comment (bew) [inline doc/manual/merging.md:270]: `command` typo?

Comment (yannham) [reply doc/manual/merging.md:270]: Good catch!
